doc/adrs/011_refactor_incremental_layout.adoc

```diff
@@ -0,0 +1,231 @@
+= ADR-011 - Refactor incremental layout
+
+== Context
+
+Following ADR-008, we implemented an incremental layout applied on each refresh.
+This algorithm follows the diagram rendering and adds layout informations on the fly, gathered either by:
+
+- previous layout informations
+- informations gathered from frontend mutations: moves, creation tools
+
+This implementation is straight-forward and simple but prevents us from anticipating crucial data.
+For instance: which size a container must have to hold all of its children, when some of them have been moved/resized.
+
+== Decision
+
+=== Process
+
+To be achieved correctly, the incremental layout must be aware of:
+
+- the whole diagram structure
+- the previous diagram layout data
+- the volatile layout data, gathered from the frontend mutations and meant to be used for a single rendering
+
+The first two items can be provided to us by the current rendering, which creates the diagram structure according to the VSM. The implementation made for ADR-008 showed that we can easily embed the previous layout data into it.
+This will constitute the first step of the rendering.
+
+The second step will consist on calling the incremental layout on the previously obtained diagram, the same way we used to call ELK automatic layout, to produce the final diagram with the layout data correctly set according to all the constraints, reconciled from the previous state with the volatile data.
+
+=== Layout Data
+
+The layout must produce the following data to achieve the incremental rendering:
+
+- the position of each node, label
+- the size of each node, label
+- the position of each bendpoint
```

Comment:
I would replace bendpoint by routing points everywhere in the document since that's the name of the concept in the code.

Reply:
Fixed in https://github.com/wpiers/sirius-components/commit/62f8cf80e053d114a0f7655bd26db3318e521999